pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -115,6 +117,36 @@ const (
 // Protects floating ip assignment
 var floatingIPAssignLock = &sync.Mutex{}
 
+func (p *provider) getConfigAuth(c *Config, rawConfig *openstacktypes.RawConfig) error {
+	var err error
+	c.ApplicationCredentialID, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.ApplicationCredentialID, "OS_APPLICATION_CREDENTIAL_ID")
+	if err == nil {
```

Comment:
`GetConfigVarStringValueOrEnv` will actually never return an error.
Thus you will always return before reading in all the non applicationCredential related variables.

This is probably causing the e2e test error because the username and password are never set.

An option would be to just remove the return command or you could check if the applicationCredentialsID is not empty

Reply:
@phiphi282 Good catch, I didn't see it. Let's see if this fixes the problem.

I would also suggest to add in a next PR an e2e test for applicationCredentials, see: 
https://github.com/kubermatic/machine-controller/blob/d7e05daefec4d9ba7a2a872ab1e4a50b8d4d08c9/test/e2e/provisioning/all_e2e_test.go#L70

e.g. OSAppCredsManifest, but that depends on the e2e test infra (OS_APPLICATION_CREDENTIAL_ID/SECRET are required)